cs/di-extensions.texy

```diff
@@ -85,61 +122,82 @@ services:
 		factory: MyBlog\Components\ArticlesList(@extension.articles)
 \--
 
-Který načteme a dodatečně služby nastavíme
+Který načteme a dodatečně služby nastavíme.
 
 /--php
-public function loadConfiguration()
+class BlogExtension extends Nette\DI\CompilerExtension
 {
-	$this->config += $this->defaults;
-	$builder = $this->getContainerBuilder();
-
-	// načtení konfiguračního souboru pro rozšíření
-	$this->compiler->loadDefinitions($builder, $this->loadFromFile(__DIR__ . '/blog.neon')['services'], $this->name);
-
-	// počet článků na stránku v komponentě
-	$builder->getDefinition($this->prefix('articlesList'))
-		->addSetup('setPostsPerPage', [$this->config['postsPerPage']]);
-
-	// volitelné vypnutí komentářů
-	if (!$this->config['comments']) {
-		$builder->getDefinition($this->prefix('comments'))
-			->addSetup('disableComments');
+	public function loadConfiguration()
+	{
+		// načtení konfigurační parametrů
+		$config = $this->validateConfig($this->defaults);
+		$builder = $this->getContainerBuilder();
+
+		// načtení konfiguračního souboru pro rozšíření
+		$this->compiler->loadDefinitions($builder, $this->loadFromFile(__DIR__ . '/blog.neon')['services'], $this->name);
+
```

Comment:
Už bych tu neopakovat příklad s setPostPage a disableComments, to se týkalo jiné kapitoly.